t/unit/worker/test_consumer.py

```diff
@@ -210,18 +214,37 @@ def bp_start(*args, **kwargs):
 
         c.blueprint.start.assert_called_once_with(c)
 
-    def test_no_retry_raises_error(self):
-        self.app.conf.broker_connection_retry = False
+    def test_too_many_open_files_raises_error(self):
         c = self.get_consumer()
-        c.blueprint.start.side_effect = socket.error()
-        with pytest.raises(socket.error):
+        err = OSError()
+        err.errno = errno.EMFILE
+        c.blueprint.start.side_effect = err
+        with pytest.raises(WorkerTerminate):
             c.start()
 
     def _closer(self, c):
         def se(*args, **kwargs):
             c.blueprint.state = CLOSE
         return se
 
+    def test_blueprint_restart_when_state_not_in_stop_conditions(self):
+        c = self.get_consumer()
+
+        # ensure that WorkerShutdown is not raised
+        c.app.conf['broker_connection_retry'] = True
```

Comment:
`broker_connection_retry` should be either True or False. In either of these cases, no warnings should be emitted.